misc/monitoring.md

```diff
@@ -0,0 +1,120 @@
+# Monitoring KubeVirt components
+
+All KubeVirt system-components expose Prometheus metrics at their `/metrics`
+REST endpoint.
+
+## Custom Service Discovery
+
+Prometheus supports service discovery based on
+[Pods](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#pod)
+and
+[Endpoints](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#endpoints)
+out of the box. Both can be used to discover KubeVirt services.
+
+All Pods which expose metrics are labeled with `prometheus.kubevirt.io` and
+contain a port-definition which is called `metrics`. In the KubeVirt
+release-manifests, the default `metrics` port is `8443`.
+
+The above labels and port informations are collected by a `Service` called
+`kubevirt-prometheus-metrics`. Kuberentes automatically creates a corresponding
+`Endpoint` with an equal name:
+
+```
+$ kubectl get endpoints -n kube-system kubevirt-prometheus-metrics -o yaml
+apiVersion: v1
+kind: Endpoints
+metadata:
+  labels:
+    kubevirt.io: ""
+    prometheus.kubevirt.io: ""
+  name: kubevirt-prometheus-metrics
+  namespace: kube-system
+subsets:
+- addresses:
+  - ip: 10.244.0.5
+    nodeName: node01
+    targetRef:
+      kind: Pod
+      name: virt-handler-cjzg6
+      namespace: kube-system
+      resourceVersion: "4891"
+      uid: c67331f9-bfcf-11e8-bc54-525500d15501
+  - ip: 10.244.0.6
+  [...]
+  ports:
+  - name: metrics
+    port: 8443
+    protocol: TCP
+```
+
+By watching this endpoint for added and removed IPs to `subsets.addresses` and
+appending the `metrics` port from `subsets.ports`, it is possible to always get
+a complete list of ready-to-be-scraped Prometheus targets.
+
+## Integrating with the prometheus-operator
+
+The [prometheus-operator](https://github.com/coreos/prometheus-operator) can
+make use of the `kubevirt-prometheus-metrics` service to automatically create
+the appropriate Prometheus scrap config.
```

Comment:
scrap? scrape?

Actually, I think it might flow better without anyways:
`the appropriate Prometheus config.`